packages/bitcore-node/src/modules/ripple/p2p.ts

```diff
@@ -0,0 +1,249 @@
+    }
+    return this.invCache[type].includes(hash);
+  }
+
+  async setupListeners() {
+    const { host, port } = this.chainConfig.provider;
+    this.events.on('disconnected', async () => {
+      logger.warn(
+        `${timestamp()} | Not connected to peer: ${host}:${port || ''} | Chain: ${this.chain} | Network: ${
+          this.network
+        }`
+      );
+    });
+    this.events.on('connected', async () => {
+      this.client!.on('ledger', async () => {
+        this.sync();
+      });
+    });
+  }
+
+  async disconnect() {
+    this.disconnecting = true;
+  }
+
+  async handleReconnects() {
+    this.disconnecting = false;
+    let firstConnect = true;
+    let connected = false;
+    let disconnected = false;
+    const { host, port } = this.chainConfig.provider;
+    while (!this.disconnecting && !this.stopping) {
+      try {
+        try {
+          this.client = await this.provider.getClient(this.network);
+          connected = this.client.isConnected();
+        } catch (e) {
+          connected = false;
+        }
+        if (connected) {
+          if (disconnected || firstConnect) {
+            this.events.emit('connected');
+          }
+        } else {
+          this.client = await this.provider.getClient(this.network);
+          this.events.emit('disconnected');
+        }
+        if (disconnected && connected && !firstConnect) {
+          logger.warn(
+            `${timestamp()} | Reconnected to peer: ${host}:${port || ''} | Chain: ${this.chain} | Network: ${
+              this.network
+            }`
+          );
+        }
+        if (connected && firstConnect) {
+          firstConnect = false;
+          logger.info(
+            `${timestamp()} | Connected to peer: ${host}:${port || ''} | Chain: ${this.chain} | Network: ${
+              this.network
+            }`
+          );
+        }
+        disconnected = !connected;
+      } catch (e) {}
+      await wait(5000);
+    }
+  }
+
+  async connect() {
+    this.handleReconnects();
+  }
+
+  public async getBlock(height: number) {
+    return this.provider.getBlock({ chain: this.chain, network: this.network, blockId: height.toString() });
+  }
+
+  async sync() {
+    if (this.syncing) {
+      return false;
+    }
+    const { chain, network } = this;
+    this.syncing = true;
+
+    try {
+      const client = await this.provider.getClient(this.network);
+      let ourBestBlock = await this.provider.getLocalTip({ chain, network });
+      let chainBestBlock = await client.getLedgerVersion();
+
+      const startTime = Date.now();
+      let lastLog = Date.now();
+
+      if (!ourBestBlock) {
+        logger.info(`Starting XRP Sync @ ${chainBestBlock}`);
+        ourBestBlock = { height: chainBestBlock - 2000 } as IXrpBlock;
+      }
+      const startHeight = ourBestBlock.height;
+      let currentHeight = startHeight;
+      while (ourBestBlock.height <= chainBestBlock) {
+        currentHeight = ourBestBlock.height + 1;
```

Comment:
i think it would make sense to get current ledger version and check that against where we are
```
let chainBestBlock = await client.getLedgerVersion();
```